compiler/cgen.nim

```diff
@@ -1117,7 +1117,10 @@ proc registerModuleToMain(g: BModuleList; m: PSym) =
   var
     init = m.getInitName
     datInit = m.getDatInitName
-  addf(g.mainModProcs, "N_LIB_PRIVATE N_NIMCALL(void, $1)(void);$N", [init])
+  if sfMainModule in m.flags:
```

Comment:
I'm not sure if this is appropriate. Is there a better way to do this?

Reply:
This check is not correct, the main module is always the main module, no matter the ``--app`` switch.